src/client_side.cc

```diff
@@ -3476,9 +3475,9 @@ ConnStateData::buildFakeRequest(const HttpRequestMethod &method, SBuf &useHost,
     // Setup Http::Request object. Maybe should be replaced by a call to (modified)
     // clientProcessRequest
     HttpRequest::Pointer request = new HttpRequest(mx);
-    AnyP::ProtocolType proto = (method == Http::METHOD_NONE) ? AnyP::PROTO_AUTHORITY_FORM : AnyP::PROTO_HTTP;
+    AnyP::ProtocolType proto = AnyP::PROTO_HTTP;
     request->url.setScheme(proto, nullptr);
-    request->method = method;
+    request->method = Http::METHOD_CONNECT;
```

Comment:
If we are using CONNECT, should not setScheme() get  `PROTO_AUTHORITY_FORM` rather than `PROTO_HTTP`? The `proto` variable also has an effect on whether the request gets a Host header below. This seems like an important decision/change. Please make sure we are not accidentally breaking things here.

Reply:
Well the CONNECT is HTTP. In any case the  HttpRequest related code handles the CONNECT with the same way the PROTO_AUTHORITY_FORM is used.

Reply:
> Well the CONNECT is HTTP. 

Yes, bit I am not sure why you are saying that. The concern here is the correct `proto` setting in this particular context. We are not setting the communication protocol here (and the communication protocol with the origin server is TCP in this case, not HTTP).

> In any case the HttpRequest related code handles the CONNECT with the same way the PROTO_AUTHORITY_FORM is used.

Sorry, I am not sure what you mean. The official code we are modifying here was using both proto values AFAICT. We have to pick one. You picked `PROTO_HTTP` but that seems to contradict the request URI format for (now hard-coded) `CONNECT` method -- `CONNECT` format [must](https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.6) follow the authority form, right? Will using `PROTO_AUTHORITY_FORM` instead of `PROTO_HTTP` here break something?

* If not, let's use that because it matches the request method.
* If it will break something, let's discuss.

Reply:
> Sorry, I am not sure what you mean. The official code we are modifying here was using both proto values AFAICT. We have to pick one.

The official code we are changing using the AnyP::PROTO_AUTHORITY_FORM as protocol if we are using Http::METHOD_NONE which means non HTTP method.

> You picked PROTO_HTTP but that seems to contradict the request URI format for (now hard-coded) CONNECT method -- CONNECT format must follow the authority form, right?

Yes. The method which computes the URI is implemented as follows:
```
const SBuf &
HttpRequest::effectiveRequestUri() const
{
    if (method.id() == Http::METHOD_CONNECT || url.getScheme() == AnyP::PROTO_AUTHORITY_FORM)
        return url.authority(true); // host:port
    return url.absolute();
}
```

  > Will using PROTO_AUTHORITY_FORM instead of PROTO_HTTP here break something?

Nope.

Reply:
Please use `PROTO_AUTHORITY_FORM` because it matches the request method and does not break anything. I (still) see no reason to raise questions about the apparent `proto`/`method` mismatch created or exposed by our changes.

Reply:
commit d0e246c8